### DISCUSSION TITLE
.NET downgrade

### DISCUSSION THREAD

**Discussion by armbrat** [nguyenquyhy/Flight-Recorder#76] [General]
First off, great app.  I really appreciate the work you've put into this.  

I pulled latest with a thought of contributing a map integration but I'm having trouble due to the UI targeting .NET 5.  Considering this is for Windows only, and is a GUI, had you considered using .NET Framework instead of .NET 5?  There are many more controls available for .NET Framework.  The only map control I could find for .NET 5 are from DevExpress and similar vendors.

**Comment by nguyenquyhy** [comment]
I don't think I would ever go back to .NET Framework; I just have too much bad experience with it and see very little benefit.

First of all, that .NET Framework is always available on Windows is basically a lie. So many times I have to go through the frustration of helping customers troubleshooting corrupted .NET Framework, wrongly reported version, or other stuffs along that line. With bundled .NET Core, I have no complaint so far, and I really appreciate that consistency.

.NET Framework is also dead, so I don't expect any issues to get fixed nor any new features. I believe DevExpress will move eventually.
.NET Core (which is now .NET) on the other hand has a lot of great features from both the new platform and new language version. I utilize those in Flight Recorder, which means .NET framework can't even compile current code.

Please understand that I am in no way not appreciate your contribution; I just can't go back to .NET Framework myself. 
Regarding the map, my plan is to leverage the new webview2. I haven't sorted out all the details yet, but I think it should be possible.

**Comment by armbrat** [comment]
I think your observations are correct.  I had been writing Windows Forms applications for nearly a decade, having only stepped away from it in the past two, happily moving on to backend service work.  Only after you bring it up do I remember all the troubles we had dealing with various frameworks on our client's machines.

I think the solution is for me to learn more about how to do Windows application development using WPF and .NET Core.  Thank you!

**Reply by nguyenquyhy** [reply]
Windows Form comes to .NET 5 too, which is awesome.

**Reply by ArcanePython** [reply]
It's there.. and I tend to agree with you about .NET framework. This myriad of versions ! Let's hope Microsoft will proceed on the path to unify things.

Really appreciate your tool as a MSFS user! And I also like to study the sources.. this statemachine approach, the code generators attached to transitions.. More work to generalize things like that, but it will pay off when more protocol will be handled !
